server/fleet/app.go

```diff
@@ -452,6 +452,8 @@ func (e *EnrollSecret) AuthzType() string {
 const (
 	EnrollSecretKind          = "enroll_secret"
 	EnrollSecretDefaultLength = 24
+	// Maximum number of enroll secrets that can be set per team, or globally.
+	MaxEnrollSecretsCount = 1000
```

Comment:
Why is this configuration necessary?

Reply:
We do not enforce any limit currently on the number of enroll secrets. Both the frontend and the backend assume a small number (e.g. no pagination, no batches of inserts, etc.) so we should enforce one, even if much higher than anything reasonable. Discussed on the ticket, awaiting feedback from Noah for the exact number (or any concerns in enforcing this). https://github.com/fleetdm/fleet/issues/7878#issuecomment-1265514585